src/Tribe/Plugin_Dependencies.php

```diff
@@ -0,0 +1,183 @@
+<?php defined( 'WPINC' ) or die;
+
+if (! class_exists( 'Tribe__Plugin_Dependencies' ) ) {
+	/**
+	 * Tracks which tribe plugins are currently activated
+	 */
+	class Tribe__Plugin_Dependencies {
+
+		/**
+		 * An multidimensional array of active tribe plugins in the following format
+		 *
+		 * array(
+		 *  'class'   => 'main class name',
+		 *  'version' => 'version num',
+		 *  'path'    => 'Path to the main plugin/bootstrap file'
+		 * )
+		 */
+		protected $active_plugins = array();
+
+		protected static $instance;
```

Comment:
For the `singleton` patter, could you follow the following: https://github.com/moderntribe/the-events-calendar/blob/develop/src/Tribe/Aggregator/Cron.php#L30-L48